examples/sse/src/app.py

```diff
@@ -0,0 +1,53 @@
+from .streaming.streams import stream_engine, stream_factory
+from fastapi import FastAPI, Request
+from sse_starlette.sse import EventSourceResponse
+from starlette.datastructures import Address
+
+import asyncio
+
+
+def create_app() -> FastAPI:
+    app = FastAPI(title="streaming-sse")
+    _setup_routes(app)
+    _on_startup(app)
+
+    return app
+
+
+async def event_publisher(*, client: Address, topic: str, group_id: str):
+    print(f"Client connected {client}")
+    stream = stream_factory(topic=topic, group_id=group_id)
+
+    async with stream as stream_flow:
+        try:
+            async for event in stream_flow:
+                yield dict(data=event)
+        except asyncio.CancelledError:
+            print(f"Disconnected from client (via refresh/close) {client}")
+
+
+def _setup_routes(app: FastAPI) -> None:
```

Comment:
I find this pattern quite strange, what's the reason to wrap everything around functions that have `app: FastAPI` as a parameter?